sdk/core/azure-core-http/src/main/java/com/azure/android/core/http/util/UrlBuilder.java

```diff
@@ -14,6 +14,7 @@
  */
 public final class UrlBuilder {
     private static final Map<String, UrlBuilder> PARSED_URLS = new HashMap<>();
+    private static final int MAX_CACHE_SIZE = 10000;
```

Comment:
@anuchandy should we lower the maximum cache size given the more restrictive environment Android generally runs on

Reply:
I don't know how often users would make use of this cache for new URLs. Do we have numbers on what normally happens in the Java SDK?

Reply:
At least in the Java SDKs this is used in RestProxy to generated the injected request URL. For many services that use the same base URL this doesn't grow often, but services like Storage where each blob has a unique base URL this is used heavily and has resulted in OOMs in the past.

Reply:
Sizing cache is not easy; here, as Alan pointed out, it depends on how likely specific services (that RestProxy proxies) tend to reuse the URLs. I'm fine to start with this number.